tests/tests_suite_test.go

```diff
@@ -122,12 +126,15 @@ var _ = BeforeSuite(func() {
 		"#!/bin/sh\nSSH_ORIGINAL_COMMAND=\"ssh $@\"\nexec /usr/bin/ssh %s -i %s \"$@\"\n",
 		sshFlags, keyPath)), 0777)
 
-	sess, err = start("deis keys:add %s.pub", keyPath)
+	pubKeyPath := keyPath + ".pub"
+	keyBytes, err := ioutil.ReadFile(pubKeyPath)
 	Expect(err).To(BeNil())
-	Eventually(sess).Should(Exit(0))
-	Eventually(sess).Should(Say("Uploading %s.pub to deis... done", keyName))
-
-	time.Sleep(5 * time.Second) // wait for ssh key to propagate
+	pubKeyInfo, err := client.ParseSSHPubKey(filepath.Base(keyPath+".pub"), keyBytes)
+	Expect(err).To(BeNil())
+	_, err = client.NewKey(testClient, pubKeyInfo.ID, pubKeyInfo.Public)
+	Expect(err).To(BeNil())
+	// TODO: not sure we need this anymore since the builder checks for valid keys on-demand.
```

Comment:
note that this is being tracked at https://github.com/deis/workflow-e2e/issues/139